core/contracts/Voting.sol

```diff
@@ -414,30 +402,14 @@ contract Voting is Testable, MultiRole, OracleInterface, VotingInterface, Encryp
         inflationRate = _inflationRate;
     }
 
-    function retrieveRewards() public {
+    function retrieveRewards(uint roundId, PendingRequest[] memory toRetrieve) public {
         uint blockTime = getCurrentTime();
-        uint roundId = votersLastRound[msg.sender];
-
-        if (roundId == voteTiming.computeCurrentRoundId(blockTime)) {
-            // If the last round the voter participated in is the current round, rewards cannot be dispatched until the
-            // round is over.
-            return;
-        }
-
-        // Round must be updated (if possible) for the voter to retrieve rewards.
-        // Note: this could be done only when the voter is trying to retrieve a reward for the most recently completed
-        // round, but it makes the logic a bit simpler to just do it in all cases.
         _updateRound(blockTime);
+        uint currentRoundId = voteTiming.computeCurrentRoundId(blockTime);
+        require(roundId < currentRoundId);
```

Comment:
nit: wdyt about adding an error message here?

Reply:
Causes out of gas exception. I think it makes the contract code too large?